### ISSUE TITLE
How can I listen the process of animation when loading gif or webp？

### ISSUE THREAD

**Issue by kris520** [facebook/fresco#1662]
Because I want to add some operation when animation processing to finish once.

**Comment by oprisnik** [comment]
You want to listen to the animation repeat event (when it jumps back to the first frame)?
Unfortunately, that's currently not supported. However, this should be ready soon since we're improving the animation API.

**Comment by kris520** [comment]
Well, I'm looking forward to.

**Comment by oprisnik** [comment]
We've released [Fresco v1.4.0](https://github.com/facebook/fresco/releases/tag/v1.4.0), which includes a new implementation for animated GIFs and animated WebPs.

You can now set  an [animation listener](https://github.com/facebook/fresco/blob/master/animated-drawable/src/main/java/com/facebook/fresco/animation/drawable/AnimationListener.java) to get notified about these events.

You can use a `ControllerListener` to attach the `AnimationListener`:
```
Fresco.newDraweeControllerBuilder()
      .setImageRequest(imageRequest)
      .setAutoPlayAnimations(true)
      .setControllerListener(
              new BaseControllerListener() {
                  @Override
                  public void onFinalImageSet(
                          String id, 
                          @Nullable Object imageInfo, 
                          @Nullable Animatable animatable) {
                      if (animatable != null) {
                          AnimatedDrawable2 animatedDrawable = (AnimatedDrawable2) animatable;
                          animatedDrawable.setAnimationListener(yourAnimationListener);
                      }
                  }
              })
      .build()
```
